### PR TITLE
UI: Add option to auto remux

### PR DESCRIPTION
**Pull request by cg2121**
This is a refactor of #908

### CODE REVIEW

**Comment by jp9000** [comment]
Hm, I'm not sure how I feel about this being directly in the normal recording settings.  The reason is because most new users won't understand this option or why it's necessary.  I think this is more of an advanced settings option.  I think it would be best if moved there instead.  What does everyone else think?

Additionally, if it's set to MP4, I feel it should record as FLV rather than as MKV.  FLV is always the safest file type due to how it works.

**Comment by jp9000** [comment]
Alright, I double-checked MKV and it's safe to use for remux.  Works great.

Last nitpick however: would prefer this in advanced settings -- off by default.  This is mostly a user experience decision.  The program defaults to FLV anyway.

**Comment by cg2121** [comment]
I have now moved the option to the advanced settings.

**Comment by cg2121** [comment]
I have now implemented your suggestions.

File: UI/obs-app.cpp [hunk 1]
```diff
@@ -77,7 +77,11 @@ string opt_starting_collection;
 string opt_starting_profile;
 string opt_starting_scene;
 
+bool remuxAfterRecord = false;
+string remuxFilename;
+
 // GPU hint exports for AMD/NVIDIA laptops
+// AMD PowerXpress High Performance Flags
```
Comment (jp9000) [inline UI/obs-app.cpp]: This line seems strangely out of place and unrelated to the commit.

File: UI/obs-app.hpp [hunk 2]
```diff
@@ -37,7 +37,7 @@ std::string CurrentTimeString();
 std::string CurrentDateTimeString();
 std::string GenerateTimeDateFilename(const char *extension, bool noSpace=false);
 std::string GenerateSpecifiedFilename(const char *extension, bool noSpace,
-					const char *format);
+		const char *format);
```
Comment (jp9000) [inline UI/obs-app.hpp]: This must be unintentional.  Probably want to remove this from the commit diff.

File: UI/window-basic-main-outputs.cpp [hunk 3]
```diff
@@ -1605,7 +1605,7 @@ bool AdvancedOutput::StartRecording()
 			strPath += "/";
 
 		strPath += GenerateSpecifiedFilename(recFormat, noSpace,
-							filenameFormat);
+					filenameFormat);
```
Comment (jp9000) [inline UI/window-basic-main-outputs.cpp]: Probably want to remove this from the commit diff as well.

**Comment by jp9000** [comment]
Looks good!  Just those few code inconsistencies and it should be good to go.  Thank you for spending the time on this.

**Comment by cg2121** [comment]
I have now got this to work with the batch remux commit.

File: UI/window-remux.cpp [hunk 4]
```diff
@@ -872,13 +888,16 @@ void OBSRemux::remuxNextEntry()
 		emit remux(inputPath, outputPath);
 	} else {
 		queueModel->endProcessing();
+
```
Comment (jp9000) [inline UI/window-remux.cpp]: Trailing whitespace here, just noticed it.

**Comment by cg2121** [comment]
The trailing white space is now removed.

**Comment by jp9000** [comment]
Great work, thanks for your patience.

**Pull request merged**